blocks/url-input/index.js

```diff
@@ -104,6 +104,12 @@ class UrlInput extends Component {
 
 	onKeyDown( event ) {
 		const { selectedSuggestion, posts } = this.state;
+		// If the suggestions are not shown, we shouldn't handle the arrow keys
+		// We shouldn't preventDefault to allow block arrow keys navigation
+		if ( ! this.state.showSuggestions || ! this.state.posts.length ) {
+			return;
+		}
```

Comment:
I'm led here via https://github.com/WordPress/gutenberg/pull/19462#issuecomment-573761895 , and trying to understand what it was that we were trying to accomplish with this condition. I'm guessing at the time it was when we were first introducing `WritingFlow` and using arrow keys to navigate between blocks. But in considering a link editor input: Why should we care whether there are suggestions present in deciding whether to prevent the default arrow key behavior? Based on what I describe in https://github.com/WordPress/gutenberg/pull/19462#issuecomment-573761895, I sorta expect that arrow keys should always use the native behavior in the input, regardless of suggestions present. I don't really expect to be able to navigate between blocks in this context.

Trying to recall: Did we used to have the link fields inline within the blocks list, and this was how we tried to allow the arrow keys to let the user go to the next/previous focusable field? Most all of those are in popovers now, correct? Maybe we don't want to _assume_ that, but I also don't think we need to optimize for WritingFlow here either.

In some ways it relates to #19488 (#18755), where I talk about what it is about these inputs which make us want them to be ignored by WritingFlow, ObserveTyping, etc. More and more, I think it's that being in a popover (modal) gives it its own context, where it should be treated as detached from the rest of the editor (and thus not inherit any of those typing behaviors).

Reply:
> I sorta expect that arrow keys should always use the native behavior in the input, regardless of suggestions present.

If URLInput is not used in the popover (Button block at that time), it shouldn't be "an arrow navigation trap"

Reply:
> If URLInput is not used in the popover (Button block at that time), it shouldn't be "an arrow navigation trap"

Yeah, thinking about it more, it feels like "popover" is more the exception (like previously mentioned with related #19488, #18755). It has me wondering whether it would be reasonable to treat popovers as entirely separate context, and to (by default) stop all event bubbling propagation from escaping. Maybe that's an extreme approach, but I kinda see it making sense as a totally independent document, almost like an `iframe`.